### DISCUSSION TITLE
How to seamlessly migrate from next-i18next to next-intl?

### DISCUSSION THREAD

**Discussion by patrikzita** [amannn/next-intl#1476] [Q&A]
Hi there, 👋

I'm currently using next-i18next in my Next.js project for handling internationalization (i18n), but I'm considering migrating to next-intl due to some of its features and flexibility. However, I'm unsure of the best approach for making this migration smooth and without breaking my existing functionality.

Next.js version: 14.2
I have something about 12 pages. I use pages router and I'd like to migrate to app router with next-intl. 

My goal is to have a smooth transition with minimal disruption to the existing user experience. Any help, examples, or insights from those who have experience with this would be greatly appreciated! 🙏

**Comment by amannn** [comment]
I don't have personal experience with this migration, but there are a few people on the issue tracker who have done this (e.g. https://github.com/amannn/next-intl/discussions/1306) — maybe if you have a specific question, they could help out.

Note that `next-intl` works both in the [App Router](https://next-intl-docs.vercel.app/docs/getting-started/app-router) as well as the [Pages Router](https://next-intl-docs.vercel.app/docs/getting-started/pages-router). You could consider migrating to `next-intl` in the Pages Router first and then later adopt the App Router.

**Comment by patrikzita** [comment]
Thank you, for your answer. I´d like to stay at next-i18next in Pages Router, but have only new pages in App Router. However, I find "collision" how setup next.config.js/next.config.ts, because next-i18next wants I18n object in next configuration and next-intl tempted us don´t use it there. Any recommendation how to make it work side by side?

**Reply by amannn** [reply]
Sounds pretty much like what @SLMNBJ has achieved here: https://github.com/amannn/next-intl/discussions/1306#discussioncomment-10488534

**Reply by patrikzita** [reply]
I'll try catch up his solution.. Thanks.
